server/src/main/java/org/opensearch/index/mapper/MapperService.java

```diff
@@ -261,7 +262,12 @@ public MapperService(
         this.idFieldDataEnabled = idFieldDataEnabled;
 
         if (INDEX_MAPPER_DYNAMIC_SETTING.exists(indexSettings.getSettings())) {
-            throw new IllegalArgumentException("Setting " + INDEX_MAPPER_DYNAMIC_SETTING.getKey() + " was removed after version 6.0.0");
+            deprecationLogger.deprecate(
+                index().getName() + INDEX_MAPPER_DYNAMIC_SETTING.getKey(),
+                "Index [{}] has setting [{}] that was removed after version ES 6.0.0, its value will be ignored.",
```

Comment:
Could we please change it to something like that (appearance of ES is a bit weird I think):

```suggestion
                "Index [{}] has setting [{}] that was removed in OpenSearch 1.0.0 and its value will be ignored.",
```